pkg/service/service.go

```diff
@@ -65,11 +69,13 @@ type svcInfo struct {
 	backends      []lb.Backend
 	backendByHash map[string]*lb.Backend
 
-	svcType                lb.SVCType
-	svcTrafficPolicy       lb.SVCTrafficPolicy
-	svcHealthCheckNodePort uint16
-	svcName                string
-	svcNamespace           string
+	svcType                   lb.SVCType
+	svcTrafficPolicy          lb.SVCTrafficPolicy
+	sessionAffinity           bool
+	sessionAffinityTimeoutSec uint32
```

Comment:
Is the extra bool needed? Or would it be possible to have a `sessionAffinity` where timeout could be 0? If not, wouldn't timeout>0 imply that it has a session affinity?

Reply:
We could imply, as `timeoutSeconds specifies the seconds of ClientIP type session sticky time. The value must be >0 && <=86400(for 1 day)` (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#clientipconfig-v1-core). However, my preference is to have a bool indicating whether the affinity is enabled (imho, better readability).